tuf/client_rework/updater_rework.py

```diff
@@ -213,97 +209,157 @@ def _load_root(self) -> None:
                 )
 
-                for temp_obj in mirror_download:
-                    try:
-                        verified_root = self._verify_root(temp_obj)
-                    # pylint: disable=try-except-raise
-                    except Exception:
-                        raise
+                # For each version of root iterate over the list of mirrors
+                # until an intermediate root is successfully downloaded and
+                # verified.
+                intermediate_root = self._root_mirrors_download(root_mirrors)
 
+            # Exit the loop when all mirrors have raised only 403 / 404 errors,
+            # which indicates that a bigger root version does not exist.
             except exceptions.NoWorkingMirrorError as exception:
                 for mirror_error in exception.mirror_errors.values():
+                    # Otherwise, reraise the error, because it is not a simple
+                    # HTTP error.
                     if neither_403_nor_404(mirror_error):
-                        temp_obj.close()
+                        logger.info(
+                            "Misc error for root version " + str(next_version)
+                        )
                         raise
 
+                logger.debug("HTTP error for root version " + str(next_version))
+                # If we are here, then we ran into only 403 / 404 errors, which
+                # are good reasons to suspect that the next root metadata file
+                # does not exist.
                 break
 
-        # Check for a freeze attack. The latest known time MUST be lower
-        # than the expiration timestamp in the trusted root metadata file
-        try:
-            verified_root.expires()
-        except exceptions.ExpiredMetadataError:
-            temp_obj.close()  # pylint: disable=undefined-loop-variable
-
-        # 1.9. If the timestamp and / or snapshot keys have been rotated,
-        # then delete the trusted timestamp and snapshot metadata files.
-        if self._metadata["root"].keys("timestamp") != verified_root.keys(
-            "timestamp"
-        ):
-            # FIXME: use abstract storage
-            os.remove(self._get_full_meta_name("timestamp"))
-            self._metadata["timestamp"] = {}
+        # Continue only if a newer root version is found
+        if intermediate_root is not None:
+            # Check for a freeze attack. The latest known time MUST be lower
+            # than the expiration timestamp in the trusted root metadata file
+            # TODO define which exceptions are part of the public API
+            intermediate_root.expires()
+
+            # 1.9. If the timestamp and / or snapshot keys have been rotated,
+            # then delete the trusted timestamp and snapshot metadata files.
+            if self._metadata["root"].keys(
+                "timestamp"
+            ) != intermediate_root.keys("timestamp"):
+                # FIXME: use abstract storage
```

Comment:
Could we use TODO instead of FIXME?
I did a quick search in the tuf repo and found only 1 place in the code, besides this file, where we are using those.
It seems we aren't using FIXME in our code and if (hopefully) one day we decide to fix our TODOs this will be left out.

Reply:
I don't really want to use abstract storage in the client (even if Metadata supports it) but as this part of code is hopefully going to move to bundle anyway, I'm not looking at this sort of things too closely now